File: starterkits/java/src/main/java/codes/blitz/challenge/bot/Bot.java
package codes.blitz.challenge.bot;

import codes.blitz.challenge.generated.Action;
import codes.blitz.challenge.generated.MoveDownAction;
import codes.blitz.challenge.generated.MoveLeftAction;
import codes.blitz.challenge.generated.MoveRightAction;
import codes.blitz.challenge.generated.MoveUpAction;
import codes.blitz.challenge.generated.TeamGameState;
import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class Bot {
  Random random = new Random();

  public Bot() {
    System.out.println("Initializing your super duper mega bot.");
  }

  /*
   * Here is where the magic happens. I bet you can do better ;)
   */
  public List<Action> getActions(TeamGameState gameMessage) {
    List<Action> actions = new ArrayList<>();

    List<Action> possibleActions =
        List.of(
            new MoveUpAction(), new MoveRightAction(), new MoveDownAction(), new MoveLeftAction());

    Action randomlyPickedAction = possibleActions.get(random.nextInt(possibleActions.size()));
    actions.add(randomlyPickedAction);

    // You can clearly do better than the random actions above. Have fun!!
    return actions;
  }
}

File: src/Bot.cs
namespace Application;

public class Bot
{
    public const string NAME = "Singe-O'-Matix=>4DaW1N";

    /// <summary>
    /// This method should be use to initialize some variables you will need throughout the game.
    /// </summary>
    public Bot()
    {
        Console.WriteLine("Initializing your super mega bot!");
    }

    /// <summary>
    /// Here is where the magic happens, for now the moves are random. I bet you can do better ;)
    /// </summary>
    public IEnumerable<Action> GetNextMoves(TeamGameState gameMessage)
    {
        Action[] possibleActions =
        [
            new MoveUpAction(),
            new MoveRightAction(),
            new MoveDownAction(),
            new MoveLeftAction(),
        ];

        // You can clearly do better than the random actions above. Have fun!!
        var nextMoves = Random.Shared.GetItems(possibleActions, 1);

        // On doit trouver la position des ennemis | les ennemis ont des personnalités différentes : C'est quoi les différences ?
        // On doit savoir où sont les murs | on peut se déplacer à une position bien précise, comment l'utiliser
        // On doit connaitre les espaces libres autour de nous
        // Connaitre le chemin le plus court pour arriver à un ennemi? Je sais pas si sa peut être utile
        // On doit pouvoir savoir si un ennemi est en vie ou pas    

        return nextMoves;
    }
}
